crates/hir_ty/src/diagnostics/match_check.rs

```diff
@@ -1495,6 +1495,20 @@ fn main(f: Foo) {
         );
     }
 
+    #[test]
+    fn internal_or() {
+        check_diagnostics(
+            r#"
+fn main() {
+    enum Either { A(bool), B }
+    match Either::B {
+        //^^^^^^^^^ Missing match arm
+        Either::A(true | false) => (),
```

Comment:
Not sure this is what this test intends to check

Reply:
That's there's a "missing match arm" diagnostic because `Either::B` isn't handled.